backend/static/frontend/org/riceapps/controllers/schedule_planner_controller.js

```diff
@@ -83,6 +84,25 @@ SchedulePlannerController.prototype.onCRNViewClick_ = function(event) {
 };
 
 /**
+ * Event handler; called when clear playground button is clicked. Clears all courses from the playground.
+ * @param {SchedulePlannerEvent} event
+ * @private
+ */
+SchedulePlannerController.prototype.onClearPlaygroundClick_ = function(event) {
+  if (this.userModel_ != null && event.type == SchedulePlannerEvent.Type.CLEAR_PLAYGROUND_CLICK){
+    var course_views = this.view_.getPlaygroundView().getChildren().slice();
+    for (var i = 0; i < course_views.length; i++) {
```

Comment:
Read my comment below; modify this to iterate using getChildCount() and getChildAt() which are inherited from goog.ui.Component.